[DOCS] Move transform resource definitions into APIs

lcawl opened the pull request:
This PR adds tagged sections that contain descriptions of the transform resource properties in https://github.com/elastic/elasticsearch/blob/master/docs/reference/rest-api/common-parms.asciidoc. It removes the transform resource definitions page (https://www.elastic.co/guide/en/elasticsearch/reference/master/transform-resource.html) and uses the shared tagged sections to put that information directly into the appropriate APIs.

It also adds two properties that were missing from the "get transforms API" results: `create_time` and `version`.

NOTE: As a result of these changes, users no longer need to flip between the transform resource page and the API page to find all the necessary details. However, from a maintenance stand-point, it means that when parameters or properties are added or removed, each of the affected APIs' pages must be updated appropriately.

This PR also qualifies the names of nested objects in the API reference pages, since otherwise it gets hard to figure out which parameters exist in which objects as the lists get long. 

Preview:
* http://elasticsearch_50108.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-transform.html
* http://elasticsearch_50108.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/update-transform.html
* http://elasticsearch_50108.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-transform.html
* http://elasticsearch_50108.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/preview-transform.html
* http://elasticsearch_50108.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/start-transform.html
* http://elasticsearch_50108.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/stop-transform.html

lcawl committed:
[DOCS] Move transform resource definitions into APIs

elasticmachine commented:
Pinging @elastic/es-docs (>docs)

elasticmachine commented:
Pinging @elastic/ml-core (:ml/Transform)

hendrikmuhs reviewed (commented):
> NOTE: As a result of these changes, users no longer need to flip between the transform resource page and the API page to find all the necessary details. 

This is great!

> However, from a maintenance stand-point, it means that when parameters or properties are added or removed, each of the affected APIs' pages must be updated appropriately.
> 

I wonder if GET really needs to copy the content, because it only returns what you PUT. Update is more tricky: You can not update all fields (maybe add some words about that?). 

Another observation (maybe for a followup PR):

The former data frame transform we used plural: transform**s** - after renaming we changed the wording (to be consistent with rollup) and now it's transform (singular). The docs still sometime use plural, e.g. "Create transforms".

lcawl committed:
[DOCS] Fixes sync properties

lcawl committed:
[DOCS] Fixes typo

lcawl commented on docs/reference/transform/apis/preview-transform.asciidoc:
```diff
@@ -37,24 +37,34 @@ on all the current data in the source index.
 [[preview-transform-request-body]]
 ==== {api-request-body-title}
 
+`pivot`::
```
I can add all the rest of the parameters from the create transform API and just mark them optional.  What sort of validation occurs?  It it worth summarizing?  Right now we don't seem to mention it at all

hendrikmuhs replied:
no, I think its to much detail to list every validation we do (I am working on a couple more as we had some user issues which we can prevent by validation). Create Transform has a nice short sentence:

"When the transform is created, a series of validations occur to ensure its success. "

So I suggest we keep it at this level, saying that validations are run (similar to create), but without going into detail.

lcawl replied:
That sounds reasonable.  I've added a similar statement to the update transform API's description.

lcawl committed:
[DOCS] Clarifies update transform API

lcawl committed:
[DOCS] Updates pluralization of transform API titles

lcawl commented:
> The former data frame transform we used plural: transforms - after renaming we changed the wording (to be consistent with rollup) and now it's transform (singular). The docs still sometime use plural, e.g. "Create transforms".

I've changed the titles so that only the stop and get APIs use plurals (since those are the ones that can affect or retrieve information about multiple transforms at once).

lcawl committed:
[DOCS] Clarified required and optional transform API parameters

lcawl commented on docs/reference/transform/apis/get-transform.asciidoc:
```diff
@@ -45,41 +45,104 @@ specifying `*` as the `<transform_id>`, or by omitting the `<transform_id>`.
-  (array) An array of {transform} resources, which are sorted by the `id` value in
-  ascending order. See <<transform-resource>>.
+The API returns an array of {transform} resources, which are sorted by the `id`
+value in ascending order and have the following properties:
```
I thought it was nice to have it all on one page, but I can just include the unique properties (create_time, version) and refer to the create transform API for the rest if this is preferred.

hendrikmuhs replied:
no super strong feeling, but as you said, if we add something we need to ensure to updates this page, too.

lcawl replied:
I've updated (removed) all but the unique properties.

lcawl committed:
Merge branch 'master' into transform-parms

lcawl committed:
[DOCS] Add info about validations in update transform API

lcawl committed:
[DOCS] Removes redundant properties from get transforms API

hendrikmuhs reviewed (approved):
LGTM
